schemas/internal/command.proto

```diff
@@ -64,16 +64,57 @@ message MetadataCommand {
     }
 }
 
+message AggregateWfMetrics {
+    WfSpecId wf_spec_id = 1;
+    string tenant_id = 2;
+    repeated StatusChanged changes = 3;
+}
+
+message PartitionMetrics {
+    repeated MetricsByTenant metrics_by_tenant = 1;
+}
+
+message MetricsByTenant {
+    string tenant_id = 1;
+    // workflow runs status changes
+    map<string, StatusChanges> lh_status_changes = 2;
+    // task run status changes
+    map<string, StatusChanges> task_status_changes = 3;
+}
+
+message StatusChanges {
+    repeated StatusChanged changes = 1;
+}
+
+message StatusChanged {
+    google.protobuf.Timestamp time = 1;
+    oneof status {
+        LHStatusChanged lh_status = 2;
+        TaskStatusChanged task_status = 3;
+    }
+    string principal_id = 4;
+}
+
+message LHStatusChanged {
+    optional LHStatus previous_status = 1;
+    LHStatus new_status = 2;
+}
+
+message TaskStatusChanged {
+    optional TaskStatus previous_status = 1;
+    TaskStatus new_status = 2;
+}
+
 // This is the repartition command
 message RepartitionCommandPb {
     google.protobuf.Timestamp time = 1;
     optional string command_id = 2;
 
     oneof repartition_command {
-        TaskMetricUpdatePb task_metric_update = 3;
-        WfMetricUpdatePb wf_metric_update = 4;
+        TaskMetricUpdatePb task_metric_update = 3; // deprecate??
```

Comment:
Well, if we removed `WfMetricUpdate` we can remove this too. And thank you for not adding `Pb` to the new proto's

Reply:
updated